### PR TITLE
[Admin][Inventory] Inventory index page in admin panel

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | yes |
| BC breaks?      | no |
| Related tickets |  |
| License         | MIT |

### CODE REVIEW

File: src/Sylius/Bundle/AdminBundle/Resources/views/Inventory/Index/_breadcrumb.html.twig [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{% import '@SyliusAdmin/Macro/breadcrumb.html.twig' as breadcrumb %}
+
+{% set breadcrumbs = [
+        { label: 'sylius.ui.administration'|trans, url: path('sylius_admin_dashboard') },
+        { label: 'sylius.ui.inventory'|trans },
+    ]
+%}
+
+{{ breadcrumb.crumble(breadcrumbs) }}
```
Comment (pamil) [inline src/Sylius/Bundle/AdminBundle/Resources/views/Inventory/Index/_breadcrumb.html.twig]:
```
{{ breadcrumb.crumble([
    { label: 'sylius.ui.administration'|trans, url: path('sylius_admin_dashboard') },
    { label: 'sylius.ui.inventory'|trans },
]) }}
```

WDYT?

File: src/Sylius/Behat/Context/Ui/Admin/ManagingInventoryContext.php [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    /**
+     * @When I filter
+     */
+    public function iFilter()
+    {
+        $this->indexPage->filter();
+    }
+
+    /**
+     * @Then I should see a single tracked variant(s) in the list
```
Comment (pamil) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingInventoryContext.php]: "single tracked variants"?

I find this step confusing, what about "I should see only one tracked variant in the list"?

File: features/inventory/managing_inventory/filtering_inventory_by_code.feature [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+        And the store has a product "Iron Maiden T-Shirt" with code "iron"
+        And "Iron Maiden T-Shirt" product is tracked by the inventory
+        And there are 5 units of product "Iron Maiden T-Shirt" available in the inventory
+        And the store has a product "RHCP T-Shirt" with code "rhcp"
+        And "RHCP T-Shirt" product is tracked by the inventory
+        And there are 25 units of product "RHCP T-Shirt" available in the inventory
+        And I am logged in as an administrator
+
+    @ui
+    Scenario: Filtering tracked product variants by a name
```
Comment (pamil) [inline features/inventory/managing_inventory/filtering_inventory_by_code.feature]: name -> code

File: features/inventory/managing_inventory/filtering_inventory_by_code.feature [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+        When I want to browse inventory
+        And I choose "Contains" as a filter code type
+        And I specify filter code as "iron"
+        And I filter
```
Comment (pamil) [inline features/inventory/managing_inventory/filtering_inventory_by_code.feature]: `And I filter tracked variants with code containing "iron"`?

I guess we don't need to create 3 steps just to filter product variants by code.

File: features/inventory/managing_inventory/filtering_inventory_by_name.feature [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        And "RHCP T-Shirt" product is tracked by the inventory
+        And there are 25 units of product "RHCP T-Shirt" available in the inventory
+        And I am logged in as an administrator
+
+    @ui
+    Scenario: Filtering tracked product variants by a name
+        When I want to browse inventory
+        And I choose "Contains" as a filter name type
+        And I specify filter name as "RHCP"
+        And I filter
```
Comment (pamil) [inline features/inventory/managing_inventory/filtering_inventory_by_name.feature]: `And I filter tracked variants with name containing "RHCP"?`

File: features/inventory/managing_inventory/filtering_inventory_by_code.feature [hunk 6]
```diff
@@ -0,0 +1,24 @@
+@managing_inventory
+Feature: Filtering inventory by a code
```
Comment (pamil) [inline features/inventory/managing_inventory/filtering_inventory_by_code.feature]: I guess we don't need an article here as it's not a concrete, single code - "by ~a~ code"

File: features/inventory/managing_inventory/filtering_inventory_by_code.feature [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    In order to filter tracked product variants by a name
```
Comment (pamil) [inline features/inventory/managing_inventory/filtering_inventory_by_code.feature]: name -> code

File: src/Sylius/Behat/Context/Ui/Admin/ManagingInventoryContext.php [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+     */
+    public function iShouldSeeASingleTrackedVariantsInTheList()
+    {
+        $foundRows = $this->indexPage->countItems();
+
+        Assert::same(
+            1,
+            $foundRows,
+            '%s rows with tracked product variants should appear on page, %s rows has been found'
```
Comment (Zales0123) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingInventoryContext.php]: Missing dot at the end of message.

File: src/Sylius/Behat/Context/Ui/Admin/ManagingInventoryContext.php [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+        $this->indexPage->specifyFilterType('name', $type);
+    }
+
+    /**
+     * @When I choose :type as a filter code type
+     */
+    public function iChooseTypeAsAFilterCodeType($type)
+    {
+        $this->indexPage->specifyFilterType('code', $type);
+    }
```
Comment (Zales0123) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingInventoryContext.php]: Why not merge this 4 definitions in 2?
```php
/**
 * @When /^I specify filter (name|code) as "([^"]+)"$/
 */
public function iSpecifyFilter($filter, $name)
{
    $this->indexPage->specifyFilterValue($filter, $name);
}

/**
 * @When I choose "([^"]+)" as a filter (code|name) type$/
 */
public function iChooseTypeAsAFilterType($type, $filter)
{
    $this->indexPage->specifyFilterType($filter, $type);
}
```

File: src/Sylius/Bundle/AdminBundle/Resources/config/grids/inventory.yml [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+                                parameters:
+                                    id: resource.product.id
+                    update:
+                        type: update
+                        options:
+                            link:
+                                parameters:
+                                    id: resource.id
+                                    productId: resource.product.id
+                    delete:
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/AdminBundle/Resources/config/grids/inventory.yml]: I'd remove this button from here.

File: src/Sylius/Bundle/AdminBundle/Resources/config/routing/inventory.yml [hunk 11]
```diff
@@ -0,0 +1,16 @@
+sylius_admin_inventory_index:
+    path: /
+    methods: [GET]
+    defaults:
+        _controller: sylius.controller.product_variant:indexAction
+        _sylius:
+            template: "@SyliusAdmin/Crud/index.html.twig"
+            grid: sylius_admin_inventory
+            section: admin
+            permission: true
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/AdminBundle/Resources/config/routing/inventory.yml]: Not needed anymore.

File: src/Sylius/Bundle/CoreBundle/Doctrine/ORM/ProductVariantRepository.php [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+
+/**
+ * @author Grzegorz Sadowski <grzegorz.sadowski@lakion.com>
+ */
+class ProductVariantRepository extends BaseProductVariantRepository implements ProductVariantRepositoryInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function createQueryBuilderByTracked($tracked = true)
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/CoreBundle/Doctrine/ORM/ProductVariantRepository.php]: createInventoryListQueryBuilder() (without the argument tracked, should be true by default)

**Comment by pjedrzejewski** [comment]
Thank you Grzesiu!

**Pull request merged**